rescompy/rescompy.py

```diff
@@ -970,37 +970,53 @@ def predict(
                               "to infer predict_length."
                         logging.warning(msg)
 
-        # If no initial state is provided, use the end of the training result.
-        if initial_state is None and resync_signal is None:
-            if isinstance(train_result, TrainResult):
-                initial_state = train_result.states[-1][None]
-                initial_input = train_result.inputs[-1][None]
-            else:
-                if initial_state is None:
-                    msg = "Must provide a TrainResult object for " \
-                          "train_result, or provide an " \
-                          "initial_state, or provide a resync_signal."
-                    logging.error(msg)
-                    
-        # If a resync signal is given, reset to all zero state and drive it
-        # with the resync signal to calculate the initial state.
-        elif initial_state is None and resync_signal is not None:
-            resync_states = self._get_states(np.zeros(self.size),
-                                             resync_signal)
-            # If an initial was provided, raise a warning that it will be
-            # ignored.
-            if initial_state is None:
+        # If resync signal is provided, use to calculate initial state and
+        # initial input.
+        if resync_signal is not None:
+            
+            # If initial_input was also explicitly provided, raise warning that
+            # it will be ignored.
+            if initial_state is not None:
                 msg = "Both initial_state and resync_signal are " \
                       "provided; initial_state will be calculated " \
                       "from resync_signal and the provided " \
                       "initial_state will be ignored."
                 logging.warning(msg)
+                
+            resync_states = self._get_states(np.zeros(self.size),
+                                             resync_signal)
             initial_state = resync_states[-1][None]
             initial_input = resync_signal[-1][None]
+
+        else:
```

Comment:
```suggestion
   elif initial_state is not None:
   ....
   elif isinstance(train_result, TrainResult):
   ...
   else:
     maybe this should raise a ValueError rather than just log the error?
```